### PR TITLE
Helm chart:  Add support for Istio mTLS

### PR DESCRIPTION
**Pull request by stefanprodan**
## Description

This PR adds the `istio.mtls` option to the Helm chart. When enabled, creates Istio policies and destination rules to enforce mTLS to all OpenFaaS components and functions excluding NATS.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Fix: #473 

## How Has This Been Tested?

Autoscaling and asyc invocation was tested on GKE v1.13 and Istio v1.2.3 with Istio gateway exposing openfaas with LE TLS.

Sync invocation LE TLS -> Istio Gateway -> mTLS -> OF Gateway -> mTLS -> Function:

Screenshot 2019-08-07 at 21 55 17

Async invocation mTLS for worker -> function and plain TCP for worker -> nats and gateway -> nats:

Screenshot 2019-08-07 at 22 49 04

The OpenFaaS gateway accepts only mTLS inbound traffic:

Screenshot 2019-08-07 at 21 58 27

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: chart/openfaas/README.md [hunk 1]
```diff
@@ -179,7 +179,7 @@ Now [verify your installation](#verify-the-installation).
 You can run the following command from within the `faas-netes/chart` folder in the `faas-netes` repo.
 
 ```sh
-helm upgrade --install openfaas openfaas/ \
+helm upgrade --install openfaas openfaas/openfaas \
```
Comment (alexellis) [inline chart/openfaas/README.md]: Did we have a typo earlier?

File: chart/openfaas/README.md [hunk 2]
```diff
@@ -219,6 +219,23 @@ By default services will be exposed with following hostnames (can be changed, se
 
 If you require TLS/SSL then please make use of an IngressController. A full guide is provided to [enable TLS for the OpenFaaS Gateway using cert-manager and Let's Encrypt](https://docs.openfaas.com/reference/ssl/kubernetes-with-cert-manager/).
 
+### Istio mTLS
```
Comment (alexellis) [inline chart/openfaas/README.md]: (Should we mention an Istio version that it's tested with?)

File: chart/openfaas/README.md (hunk 1 above)
Comment (stefanprodan) [reply chart/openfaas/README.md]: yeap

File: chart/openfaas/README.md [hunk 3]
```diff
@@ -219,6 +219,23 @@ By default services will be exposed with following hostnames (can be changed, se
...
+
+To install OpenFaaS with Istio mTLS pass  `--set istio.mtls=true` and disable the HTTP probes:
```
Comment (alexellis) [inline chart/openfaas/README.md]: I think you mentioned this forgot, sorry (I forgot) but are httpProbes available for newer versions of Istio, or not available for any?

Comment (stefanprodan) [reply chart/openfaas/README.md]: I've tried to enable those on Istio 1.2.3 and it didn't work.

File: chart/openfaas/README.md (hunk 2 above)
Comment (stefanprodan) [reply chart/openfaas/README.md]: Done

File: chart/openfaas/README.md (hunk 3 above)
Comment (alexellis) [reply chart/openfaas/README.md]: I'll merge as it, but can you look into this when you have time?

**Comment by alexellis** [comment]
Thank you @stefanprodan 🥇

**Pull request merged**